### DISCUSSION TITLE
guides/conda-quickstart/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8738] [Docs]
# guides/conda-quickstart/

Comprehensive guide to setting up and using Ultralytics YOLO models in a Conda environment. Learn how to install the package, manage dependencies, and get started with object detection projects.

https://docs.ultralytics.com/guides/conda-quickstart/

**Comment by nadinator** [comment]
Does the python version have to be 3.8? Can it be a newer version? What would/wouldn't break if we used 3.11, for example (I'm thinking of the [ray]tuning feature)?

**Reply by glenn-jocher** [reply]
Hey there! 😊

You can definitely use Python versions newer than 3.8, including 3.11. However, it's important to note that while most features should work fine, there might be some compatibility issues with newer Python versions, especially with external libraries or specific features like [ray] tuning.

If you're keen on using Python 3.11, I'd recommend testing the key functionalities you plan to use in your project to ensure everything runs smoothly. For the [ray] tuning feature, you might want to check Ray's documentation or GitHub issues for any known compatibility concerns with Python 3.11.

Here's a quick example of how you might set up your environment with Python 3.11:

```bash
conda create --name ultralytics-env python=3.11 -y
conda activate ultralytics-env
conda install -c conda-forge ultralytics
```

And remember, if you encounter any issues, the Ultralytics community is here to help! 🚀

**Comment by Malvinlam** [comment]
Hi is there a way to install the yolov8 dependencies using Anaconda prompt to the virtual environment other than pip install ultralytics? For example I already downloaded the ultralytics package from GitHub and opened it as project folder using PyCharm. Is there a way to only install the required libraries?

**Reply by glenn-jocher** [reply]
@Malvinlam hi there! 👋

Absolutely, you can install the required dependencies for YOLOv8 using Conda without relying on `pip install ultralytics`. Here's a step-by-step guide to help you set up your Conda environment and install the necessary libraries:

1. **Create a New Conda Environment**:
   First, create a new Conda environment with Python 3.8 (or your preferred version):
   ```bash
   conda create --name ultralytics-env python=3.8 -y
   ```

2. **Activate the Environment**:
   Activate your newly created environment:
   ```bash
   conda activate ultralytics-env
   ```

3. **Install Required Libraries**:
   You can install the required libraries from the `requirements.txt` file in the Ultralytics repository. First, navigate to the directory where your `requirements.txt` file is located, then run:
   ```bash
   conda install --file requirements.txt
   ```

   Alternatively, if you want to install specific packages manually, you can use:
   ```bash
   conda install -c conda-forge numpy pandas matplotlib opencv scikit-learn
   conda install -c pytorch pytorch torchvision torchaudio
   ```

4. **Note on CUDA Environment**:
   If you're working in a CUDA-enabled environment, it's a good practice to install `pytorch` and `pytorch-cuda` together to resolve any conflicts:
   ```bash
   conda install -c pytorch -c nvidia -c conda-forge pytorch torchvision torchaudio pytorch-cuda=11.8
   ```

5. **Using Ultralytics**:
   With the dependencies installed, you can now start using Ultralytics. For example, to perform inference with YOLOv8:
   ```python
   from ultralytics import YOLO

   model = YOLO("yolov8n.pt")  # initialize model
   results = model("path/to/image.jpg")  # perform inference
   results[0].show()  # display results for the first image
   ```

For more detailed instructions, you can refer to our [Conda Quickstart Guide](https://docs.ultralytics.com/guides/conda-quickstart/).

If you encounter any issues or have further questions, feel free to ask. Happy coding! 😊

**Comment by najwazkiyahh** [comment]
Hi! I'm the new to using YOLO.
I have python 3.12 on my laptop because my previous exam need it-the latest version of python.
I want to install ultralytics YOLOv11, should I downgrade my python version from 3.12 into 3.11?
Or I should downgrade my python version to the recommended requirement, 3.8?
Thank you, hope you (anybody) can answer my confusion.

**Reply by glenn-jocher** [reply]
@najwazkiyahh you don't need to downgrade your system Python! 😊 Simply create a Conda environment with Python 3.11 using `conda create --name yolo_env python=3.11`, then activate it and install Ultralytics. This keeps your existing Python 3.12 intact while giving you a compatible YOLO11 environment. For full steps, see the Conda Quickstart Guide for Ultralytics. 

We recommend this approach as it maintains version isolation while allowing you to work with both Python versions seamlessly. Welcome to the YOLO community! 🚀
